### PR TITLE
[java-sdk-v2] Tax category service migration

### PR DESCRIPTION
**Pull request by lojzatran**
#### Summary

Migrate TaxCategoryService to java sdk v2

### CODE REVIEW

File: src/integration-test/java/com/commercetools/sync/integration/sdk2/commons/utils/TaxCategoryITUtils.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    return ctpClient
+        .taxCategories()
+        .post(taxCategoryDraft)
+        .execute()
+        .thenApply(ApiHttpResponse::getBody)
+        .join();
+  }
+
+  /**
+   * Creates a {@link io.sphere.sdk.taxcategories.TaxRateDraft} with the name {@value
```
Comment (salander85) [inline src/integration-test/java/com/commercetools/sync/integration/sdk2/commons/utils/TaxCategoryITUtils.java:106]: Replace the link with path to the v2 model.

**Review by salander85 — CHANGES_REQUESTED** [review]

File: src/integration-test/java/com/commercetools/sync/integration/sdk2/commons/utils/TaxCategoryITUtils.java (hunk 1 above)
Comment (lojzatran) [reply src/integration-test/java/com/commercetools/sync/integration/sdk2/commons/utils/TaxCategoryITUtils.java:106]: https://github.com/commercetools/commercetools-sync-java/pull/920/commits/8f1cfc505581c5b334ea7f95faeb1a529331feff

File: src/main/java/com/commercetools/sync/sdk2/taxcategories/TaxCategorySyncOptionsBuilder.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    this.ctpClient = ctpClient;
+  }
+
+  /**
+   * Creates a new instance of {@link
+   * com.commercetools.sync.sdk2.taxcategories.TaxCategorySyncOptionsBuilder} given a {@link
+   * io.sphere.sdk.client.SphereClient} responsible for interaction with the target CTP project,
+   * with the default batch size ({@code BATCH_SIZE_DEFAULT} = 50).
+   *
+   * @param ctpClient instance of the {@link io.sphere.sdk.client.SphereClient} responsible for
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/taxcategories/TaxCategorySyncOptionsBuilder.java:30]: https://github.com/commercetools/commercetools-sync-java/pull/920/commits/8f1cfc505581c5b334ea7f95faeb1a529331feff

File: src/main/java/com/commercetools/sync/sdk2/services/TaxCategoryService.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public interface TaxCategoryService {
+
+  /**
+   * Filters out the keys which are already cached and fetches only the not-cached tax category keys
+   * from the CTP project defined in an injected {@link io.sphere.sdk.client.SphereClient} and
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/services/TaxCategoryService.java]: https://github.com/commercetools/commercetools-sync-java/pull/920/commits/8f1cfc505581c5b334ea7f95faeb1a529331feff

File: src/main/java/com/commercetools/sync/sdk2/services/impl/TaxCategoryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+    final ByProjectKeyTaxCategoriesGet query =
+        syncOptions
+            .getCtpClient()
+            .taxCategories()
+            .get()
+            .withWhere("key in :keys")
+            .withPredicateVar("keys", Collections.singletonList(key));
+
+    return fetchCachedResourceId(key, query);
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/services/impl/TaxCategoryServiceImpl.java]: https://github.com/commercetools/commercetools-sync-java/pull/920/commits/24c29f47766d0bd67c9473bca1c8c91674b49c7f

File: src/main/java/com/commercetools/sync/sdk2/services/impl/TaxCategoryServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  }
+
+  @Nonnull
+  @Override
+  public CompletionStage<Set<TaxCategory>> fetchMatchingTaxCategoriesByKeys(
+      @Nonnull final Set<String> keys) {
+    return fetchMatchingResources(
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/services/impl/TaxCategoryServiceImpl.java]: https://github.com/commercetools/commercetools-sync-java/pull/920/commits/24c29f47766d0bd67c9473bca1c8c91674b49c7f

**Review by salander85 — APPROVED** [review]

**Pull request merged**